presto-tests/src/test/java/io/prestosql/tests/TestWorkProcessorPipelineQueries.java

```diff
@@ -32,6 +32,12 @@ protected TestWorkProcessorPipelineQueries()
     @Test
     public void testTopN()
     {
-        assertQuery("select * from orders order by totalprice limit 10");
+        assertQuery("SELECT * FROM orders ORDER BY totalprice LIMIT 10");
+    }
+
+    @Test
+    public void testSemiJoin()
+    {
+        assertQuery("SELECT * FROM lineitem WHERE orderkey IN (SELECT orderkey + 1 FROM lineitem WHERE orderkey < 100) LIMIT 10");
```

Comment:
You could use more realistic query, e.g
```
SELECT AVG(extendedprice) FROM lineitem where partkey IN (SELECT partkey FROM partsupp WHERE supplycost < xyz)
```
`xyz` some value that preserves some rows
```

Also please add `NOT IN` query